configure.ac

```diff
@@ -44,59 +58,122 @@ AC_PROG_CPP
-AC_CHECK_PROGS([FMS_BATS], [bats])
-if test -z "$FMS_BATS"; then
-   AC_MSG_ERROR([Cannot find bats utility. Install bats and try again.])
+AC_CHECK_PROGS([BATS], [bats])
+AM_CONDITIONAL([MAKE_CHECK_ENABLED], [test -n "$BATS"])
+if test -z "$BATS"; then
+   AC_MSG_WARN([Cannot find the bats utility. All checks will be disabled.])
 fi
 
 # Check to see if any macros must be set to enable large (>2GB) files.
 AC_SYS_LARGEFILE
 
-# Require MPI.
-AC_CHECK_FUNC([MPI_Init], [], [AC_MSG_ERROR([MPI C library required to build FMS])])
+# C specific checks
+AC_LANG_PUSH(C)
+if test $with_mpi = yes; then
+  # Require MPI
+  # We expect users to set CC and FC to MPI compiler wrappers, or that the correct
+  # CPPFLAGS, {C,FC}FLAGS, LDFLAGS and LIBS options are given.
+  AC_CHECK_HEADERS([mpi.h], [], [AC_MSG_ERROR([Can't find the MPI C header file.  Set CC/CPPFLAGS/CFLAGS])])
+  AC_CHECK_FUNC([MPI_Init], [], [AC_MSG_ERROR([Can't find the MPI C library.  Set CC/LDFLAGS/LIBS])])
+fi
 
-# Check for netCDF C library.
-AC_SEARCH_LIBS([nc_create], [netcdf], [],
-                            [AC_MSG_ERROR([Can't find or link to the netcdf C library, set CPPFLAGS/LDFLAGS.])])
+# Require netCDF
+AC_CHECK_HEADERS([netcdf.h], [], [AC_MSG_ERROR([Can't find the netCDF C header file.  Set CPPFLAGS/CFLAGS])])
+AC_SEARCH_LIBS([nc_create], [netcdf], [], [AC_MSG_ERROR([Can't find the netCDF C library.  Set LDFLAGS/LIBS])])
 
-# Check for netCDF Fortran library.
+if test $with_openmp = yes && test $enable_setting_flags = yes; then
+  # Check if the compiler needs special OpenMP flags
+  AC_OPENMP()
+fi
+AC_LANG_POP(C)
+
+# Fortran specific checks
 AC_LANG_PUSH(Fortran)
-AC_SEARCH_LIBS([nf_create], [netcdff], [],
-                            [AC_MSG_ERROR([Can't find or link to the netcdf Fortran library, set CPPFLAGS/LDFLAGS.])])
-AC_LANG_POP(Fortran)
+if test $with_mpi = yes; then
+  # Require MPI
+  GX_FORTRAN_CHECK_HEADERS([mpif.h], [], [AC_MSG_ERROR([Can't find the MPI Fortran header file.  Set FC/CPPFLAGS/FCFLAGS])])
+  GX_FC_CHECK_MOD([mpi], [], [], [AC_MSG_ERROR([Can't find the MPI Fortran module.  Set FC/CPPFLAGS/FCFLAGS])])
+  AC_CHECK_FUNC([MPI_init], [], [AC_MSG_ERROR([Can't find the MPI Fortran library.  Set FC/LDFLAGS/LIBS])])
+fi
+
+# Require netCDF
+GX_FORTRAN_CHECK_HEADERS([netcdf.inc], [], [AC_MSG_ERROR([Can't find the netCDF Fortran header file.  Set CPPFLAGS/FCFLAGS])])
+GX_FC_CHECK_MOD([netcdf], [], [], [AC_MSG_ERROR([Can't find the netCDF Fortran module.  Set CPPFLAGS/FCFLAGS])])
+GX_FORTRAN_SEARCH_LIBS([nf_create], [netcdff], [@%:@include <netcdf.inc>], [iret = nf_create('foo.nc', 1, ncid)], [],
+  [AC_MSG_ERROR([Can't find the netCDF Fortran library.  Set LDFLAGS/LIBS])])
+GX_FORTRAN_SEARCH_LIBS([nf90_create], [netcdff], [use netcdf], [iret = nf90_create('foo.nc', 1, ncid)], [],
+  [AC_MSG_ERROR([Can't find the netCDF Fortran library.  Set LDFLAGS/LIBS])])
+
+# Check if Fortran compiler has Cray pointer support
+GX_FC_CRAY_POINTER_FLAG()
```

Comment:
If this and following functions set flags, you want to protect them by checking enable_setting_flags.

Reply:
Here that restriction is not needed.  The flag required to support cray pointer is only added to `FCFLAGS` if we are setting the flags for the user (see lines 135 and 156).

However, something should be done to catch the case where the user has not supplied the required flag(s).  That can be added in a future commit.